pkg/apis/cluster/v1alpha1/machine_types.go

```diff
@@ -151,6 +151,35 @@ type MachineStatus struct {
 	// Refer: https://kubernetes.io/docs/concepts/architecture/nodes/#condition
 	// +optional
 	Conditions []corev1.NodeCondition `json:"conditions,omitempty"`
+
+	// LastOperation describes the last-operation performed by the machine-controller.
+	// This API should be useful as a history in terms of the latest operation performed on the
+	// specific machine. It should also convey the state of the latest-operation for example if 
+	// it is still on-going, failed or completed successfully.
+	// +optional
+	LastOperation LastOperation `json:"lastOperation,omitempty"`
```

Comment:
Please make this a pointer as its an optional struct

Reply:
yes sure, done.